### ISSUE TITLE
Way to dismiss videos or mark them as "Not Interested"

### ISSUE THREAD

**Issue by lima8zulu** [FreeTubeApp/FreeTube#1362]
It would be great if there was a way to mark videos as "Not Interested" (to watch) or a way to dismiss or hide them. 

Currently I hide watched videos. If there's a video in my feed I don't want to watch I mark is as read to make it disappear, but it messes up with my watch history. It would be great if there was an option to just hide videos without marking them as watched.

**Comment by lagapi** [comment]
I believe something to easily mark a video as watched or non interesting would be very useful. The same way we have a star button at the right of every thumbnail video, it could be added another button in the left part for this. Maybe a little circle like the read/unread button from thunderbird.

**Comment by pseudomonas** [comment]
I don't want to hide watched videos (I like to re-watch) but sometimes there's a video with a thumbnail or title that I find distressing, and while on YouTube it's easy to hide, on Freetube it reqiuires workarounds. 

There might need to be some kind of interface to undo the hiding, though I guess this could be just a button in the settings that removes _all_ videos from being hidden, similar to deleting all history and marking all videos as unwatched?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Maybe a weirs suggestion but

What about adding the whole title of the video to the blacklist by clicking on a hide video button in the dropdown options

**Comment by jasonhenriquez** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc I can see why you're saying that as that would be very easy to implement after #4202, but I am a bit worried about the imperfections of that solution w.r.t. unintuitive and undesirable byproducts of it (e.g., you block a video titled "a", and now you can't see most videos, or the title is a generic one word like "Music" and you now stop seeing videos with that in the title).

**Comment by pseudomonas** [comment]
I think if I were implementing this, from looking a little at the code, I'd copy the "history"/"watched" mechanism so one could have a parallel to  `.historyEntryExists()` and  `.markAsWatched()` methods.  

Then one could hide videos on the `hidden` list as though one was hiding already-watched videos.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> @efb4f5ff-1298-471a-8973-3d47447115dc I can see why you're saying that as that would be very easy to implement after #4202, but I am a bit worried about the imperfections of that solution w.r.t. unintuitive and undesirable byproducts of it (e.g., you block a video titled "a", and now you can't see most videos, or the title is a generic one word like "Music" and you now stop seeing videos with that in the title).

What about when you press on hide video button it adds it to the blacklist based on video url?

**Comment by pseudomonas** [comment]
> What about when you press on hide video button it adds it to the blacklist based on video url? 

I can't _think_ of a way that this differs from the functionality that I'm asking for, apart from the _extremely_ tenuous edge-case of "someone has a video **title** that contains another video's **URL**" and I'm not sure I've ever seen anyone doing that. They _might_ include the other URL in the video description but blacklisting doesn't take that into account, right?

Am I right in thinking that blacklisting a video hides it both from the subscription list, and from the related-videos list?

**Comment by jasonhenriquez** [comment]
> What about when you press on hide video button it adds it to the blacklist based on video url?

Yes, saving the video ID should always work to my knowledge. We'll be checking a specific field for this, so no edge cases around titles either.

> Am I right in thinking that blacklisting a video hides it both from the subscription list, and from the related-videos list? (this seems to me the desirable behaviour.)

For blacklisting a video using this feature, yes, that would be necessary. We would have to hide it just about everywhere to achieve the same "get out of my sight" effect. One remaining question is whether we still hide it in your History or Favorites if you saved it to either of them. I'm leaning towards hiding it from History, but not Favorites. Another remaining question is if we make the video a "video hidden" list item that you can click on to unhide, either only for that very page you block it (which is how YT handles it, to my knowledge) or ad infinitum (whenever you chance upon it).

**Comment by jasonhenriquez** [comment]
Huh, apparently, YT's version of this Hide feature is a lot more limited than I'm suggesting - applies to Subscriptions and Recommended Videos, but nowhere else as far as I can tell. Not even search results. You can't do it in playlists either, but it at least seems to respect the hiding of them there. We might have to consider that in our implementation - at least w.r.t. the question of whether to hide things on Channel pages. But our feature will most likely be more expansive than theirs in terms of affected views, and views where you're presented with that choice. I also didn't see anything on where you could clear your hidden videos, which was pretty enlightening as well.

**Comment by lamborghinipth** [comment]
When will this feature be implemented?

**Comment by lamborghinipth** [comment]
No one need this feature, doesnt it?

**Comment by pseudomonas** [comment]
This software is built and maintained by volunteers. It's really bad form to nag, especially when we're talking about a desired feature rather than a bug. You can see the activity where the maintainers have moved it to "to assign" which means that they're planning to work on this when they have the available manpower.

(I''m not gonna do the "fix it yourself" cliché because most people including me don't have the skills, but please have some grace not to be passive-aggressive about it.)

**Comment by pseudomonas** [comment]
I'm confused, I thought you were nagging other people to fix this but you're one of the contributors?

**Comment by lamborghinipth** [comment]
> I'm confused, I thought you were nagging other people to fix this but you're one of the contributors?

No, I am not a contributor. Just an end user. End users don't have the right to request features, do they?

**Comment by zacryol** [comment]
Perhaps this could be two separate features: a "Dismiss" option that *only* hides the video from the Subscriptions page (simply dismissing the "notification," and not the video as a whole), and a separate "Block Video" option that hides it everywhere, including recommendations, search, and the uploader's channel page.

The simple "Dismiss" option can have advantages in the situation described by the poster of #5261 , regarding "multiple streams of content." For example, a channel may be doing spoiler-full videos on some game/show a user hasn't played/seen yet; they can dismiss those videos from the sub feed for cleanup in the short term, but the user can reasonably go back to the channel and watch those videos after finishing on their own, so fully blocking them the way they might block a video they find wholly objectionable wouldn't suffice there.

**Comment by angrymidget4728** [comment]
A tag system might be the most robust way of handling this. That way we can have more than just 'Not interested' and we can set some tags to be blacklisted so that they don't appear in our feed at all.

A quicker way of implementing might actually be to utilize playlists like this. Let us blacklist videos in particular playlists. Any video we add in those playlists will never be shown in our feed. Basically something opposite of 'Favorites'.
